application/src/main/java/com/ericsson/bss/cassandra/ecchronos/application/DefaultNativeConnectionProvider.java

```diff
@@ -98,31 +100,44 @@ private static LocalNativeConnectionProvider establishConnection(
             final LocalNativeConnectionProvider.Builder builder,
             final String host,
             final int port,
-            final long timeout)
+            final long timeout,
+            final RetryPolicy retryPolicy)
     {
-        long startTime = System.currentTimeMillis();
-        long endTime = startTime + timeout;
-        while (endTime > System.currentTimeMillis())
+        for (int count = 1; count <= retryPolicy.getMaxAttempts(); count++)
         {
             try
             {
                 return builder.build();
             }
             catch (AllNodesFailedException | IllegalStateException e)
             {
+                LOG.warn("Unable to connect through CQL using {}:{}, retrying", host, port);
+                long delay = retryPolicy.currentDelay(count);
+                long currentTime = System.currentTimeMillis();
+                long endTime = currentTime + timeout;
                 try
                 {
-                    LOG.warn("Unable to connect through CQL using {}:{}, retrying", host, port);
-                    if (timeout != 0)
+                    if (currentTime + delay > endTime)
                     {
-                        LOG.debug("Connection failed, retrying in 5 seconds", e);
+                        String msg = String.format(
+                                "Connection failed in attempt %d of %d. Retrying in 5 seconds.",
```

Comment:
Hardcoded 5. Replace with a %d instead and adding the sleep time, just in case the sleep time is changed later on.